Infra/keycloak helm kustomize

Guy-Ghis opened the pull request:
Prepared production ready helm charts and kustomize manifests with proper documentation to deploy keycloak to a kubernetes cluster.

Guy-Ghis committed:
fix: update health probe port from 8080 to 9000

- Keycloak 26.4.0 exposes health endpoints on management port 9000
- Updated liveness, readiness, and startup probes
- Fixes pod readiness detection issues
- Applied to both Helm and Kustomize deployments

Guy-Ghis committed:
fix: configure postgres service names in kustomize overlays

- Fixed dev overlay to patch KC_DB_URL_HOST (env index 3) instead of KC_DB
- Added postgres-service-staging patch for staging overlay
- Added postgres-service-prod patch for production overlay
- Resolves CrashLoopBackOff in dev and prod environments

Guy-Ghis committed:
docs: add production security checklist

- Created SECURITY.md with production readiness guidelines
- Documents critical security requirements (credentials, TLS, persistence)
- Includes production deployment checklist
- Replaces deleted PRODUCTION_READINESS.md

Guy-Ghis committed:
docs: update deployment guides for kustomize migration

- Updated README with kustomize as primary deployment method
- Added health probe port 9000 documentation
- Documented postgres storage configuration (ephemeral by default)
- Added environment-specific postgres service names
- Updated local deployment instructions with multiple options
- Removed references to deleted postgres-statefulset.yaml
- Added quick start sections for testing

Guy-Ghis committed:
feat: add Keycloak Helm chart with configuration, templates, and documentation

Guy-Ghis committed:
feat: add Keycloak deployment configuration with ingress, services, and database setup

Guy-Ghis committed:
feat: add initial kind configuration for Kubernetes cluster setup

Guy-Ghis committed:
feat: add script for creating Keycloak secrets and Helm deployment guide

Nkwenti-Severian-Ndongtsop commented on README.md:
```diff
@@ -1 +1,571 @@
+
+**All files have been updated to Keycloak 26.4.0**. To complete the setup, create two additional template files:
+
+### Required New Files:
```
@Guy-Ghis you said we need to create this files, but going to where they were to be created, the files were already present
I will like you to check that ??

Guy-Ghis replied:
You are right, in this case I will just update the readme to remove that section cause the files on their own do not pose any problem because they can be enabled, disabled, or customised in the deployment depending on the needs of the user.

Guy-Ghis committed:
docs: update README to remove references to additional template files

Guy-Ghis committed:
feat: add NetworkPolicy and ConfigMap for kustomize deployment

Guy-Ghis committed:
docs: update and reorganize prerequisites and infrastructure requirements in README

Guy-Ghis committed:
feat: add database password configuration for Keycloak Helm installation commands

Guy-Ghis committed:
fix: add missing namespace for PostgreSQL resources in kustomize configuration

Guy-Ghis committed:
docs: update README local development section to meet updates

Guy-Ghis commented on README.md:
```diff
@@ -1 +1,571 @@
+- **Helm Charts**: Full-featured, configurable Helm chart for flexible deployments
+
+**✅ Status:** All deployments tested and working with Keycloak 26.4.0
+
```
Sure, that makes perfect sense
I’ll add a section with the required prerequisites (like Docker, kubectl, Helm, and Kustomize) along with links to their official installation guides for different OSes, including macOS. Thanks for the suggestion.

Guy-Ghis commented on README.md:
```diff
@@ -1 +1,571 @@
+kubectl get pods -n keycloak
+
+# Access Keycloak (port-forward for testing)
+kubectl port-forward -n keycloak svc/keycloak 8080:8080
```
The error was caused by the Keycloak deployment not being able to bind with the PostgreSQL database, either because it was deployed previously, in a different namespace from that of the Keycloak deployment, or the password used inside the command did not match the one set inside the postgres.YAML file. The issue has been addressed and the documentation updated accordingly.

Guy-Ghis committed:
refactor(docs): streamline and clean documentation structure but no functional change

marcjazz reviewed (commented):
Please confirm that bitnami helm chart release for keycloak is no longer available for the public fier tier. 

Also document the decision to create our own chart if it comes to it.

Nkwenti-Severian-Ndongtsop reviewed (changes requested):
@Guy-Ghis the changes were good, i was able to provision the keycloak using the README.md file, so you're left just resolving your conflict for me to approve the changes

Guy-Ghis committed:
Remove deprecated Keycloak Helm chart and Kustomize configurations

- Deleted HPA, Ingress, NetworkPolicy, Secret, Service, ServiceAccount templates from Helm chart.
- Removed PostgreSQL configuration and related resources from Kustomize base.
- Eliminated local development and testing files, including Kind configuration and README.
- Cleaned up Kustomize overlays for dev, staging, and production environments.
- Updated values.yaml to remove unused configurations and secrets.

Guy-Ghis committed:
feat: add comprehensive changelog and technical guide documentation for Keycloak deployment

Guy-Ghis committed:
Refactor code structure for improved readability and maintainability

Guy-Ghis committed:
feat: add Kustomize deployment files for Keycloak and PostgreSQL setup

Guy-Ghis committed:
Merge branch 'main' of https://github.com/ADORSYS-GIS/keycloak-ops into infra/keycloak-helm-kustomize

Guy-Ghis committed:
feat: add README.md for Keycloak deployment with Bitnami Helm charts

Nkwenti-Severian-Ndongtsop reviewed (commented):
@Guy-Ghis the implementation looks good and works for me
but i don't know if it configurable, like changing the default username, the db ??

Nkwenti-Severian-Ndongtsop commented:
@Guy-Ghis i will love us to have a demo for this, because we have a lot of files so you can like explain the necessity of files and how they help in the deployment process

Guy-Ghis commented:
> @Guy-Ghis i will love us to have a demo for this, because we have a lot of files so you can like explain the necessity of files and how they help in the deployment process
Yes @Nkwenti-Severian-Ndongtsop everything is fully configurable and I will make a demo during tomorrow's meeting and explain everything.

Guy-Ghis committed:
Fix: Improve infrastructure document

JOELNATHAN544 reviewed (commented):
I tested it for both helm and kustomize and found that the  initialDelaySeconds that is 120 s in the values-dev.yaml was too small and causing the pod to restart each time and I had to increase it to 300s before the pod could now be created fully and ready. So it is best to increase the initialDelaySeconds time...
